linode/manager
# fix: [M3-8602] - Fix `plan-selection.spec.ts` test failures on retries

## Description 📝

This fixes a minor test flaw in the `plan-selection.spec.ts` spec where mocks are getting set up in `before` hooks instead of `beforeEach` hooks. This prevents mocks from being applied on test retries (because mocks get cleared between tests, but `before` hooks only get called once), causing retries in this spec to be guaranteed to fail.

## Changes  🔄
- Replace `before` hooks with `beforeEach` in `plan-selection.spec.ts` to ensure mocks get applied on test retries

## Target release date 🗓️
N/A

## How to test 🧪

### Reproduction steps
In `develop` (edit: the issue causing the initial failure has been fixed, so reproducing this is now pretty difficult), build Cloud via `yarn && yarn build && yarn start:manager:ci` and then:

```bash
repeat 5 yarn cy:run -s "cypress/e2e/core/linodes/plan-selection.spec.ts
```

Observe flakiness (in my case, the spec failed 4 out of 5 times).

### Verification steps
Check out this branch, build Cloud via `yarn && yarn build && yarn start:manager:ci` and then:

```bash
repeat 5 yarn cy:run -s "cypress/e2e/core/linodes/plan-selection.spec.ts
```

Observe the test passes each time. I was able to run the test 50 times on repeat without failures.

We can also rely on CI to some extent -- this issue theoretically impacts all of our Autocompletes, so any test failures involving an Autocomplete (including region selects, etc.) should receive extra scrutiny.

### Manual verification
Some extra attention should be given to the "Phone Verification" field at `/profile/auth` since it's the only Autocomplete I'm aware of that takes advantage of the recent changes to the `PopperComponent` prop, so it may be at more risk of being broken by these changes than other Autocompletes.

## As an Author I have considered 🤔

*Check all that apply*

  - Pushed a changeset for the test fix since that's been a flaw all along, but don't think a changeset is necessary for the `Autocomplete` change since 10780 hasn't been released yet. Thoughts?
